torch/utils/_sympy/value_ranges.py

```diff
@@ -309,6 +309,22 @@ def floordiv(a, b):
         else:
             return ValueRanges.coordinatewise_monotone_map(a, b, operator.floordiv)
 
+    @staticmethod
+    def expr_cond_pair(a, b):
+        # This handles a pair (expr, cond), where expr is a SymPy expression,
+        # cond represents when it's. The output range is just expr's range.
```

Comment:
when it's... ?

This seems like a pretty odd thing to have put on value ranges, I would have expected you to have handled this in piecewise all in one go.

Reply:
This is kind of implementation detail of piecewise in sympy [here](https://github.com/sympy/sympy/blob/d2be7bacd2604e98a642f74028e8f0d7d6084f78/sympy/functions/elementary/piecewise.py#L137),  which calls this expr_cond_pair inside.  The value ranges for expr_cond_pair will be evaluated before the evaluation of piecewise. So I cound't do it all in one go in piecewise.

Should we keep the current implementation? Or should I figure out a way to avoid adding the expr_cond_pair method.